client/state/utils.js

```diff
@@ -116,3 +194,45 @@ export function createReducer( initialState = null, customHandlers = {}, schema
 		return state;
 	};
 }
+
+/**
+ * Creates schema-validating reducer
+ *
+ * Use this to wrap simple reducers with a schema-based
+ * validation check when loading the initial state from
+ * persistent storage.
+ *
+ * When this wraps a reducer with a known JSON schema,
+ * it will intercept the DESERIALIZE action (on app boot)
+ * and check if the persisted state is still valid state.
+ * If so it will return the persisted state, otherwise
+ * it will return the initial state computed from
+ * passing the null action.
+ *
+ * @example
+ * const ageReducer = ( state = 0, action ) =>
+ *     GROW === action.type
+ *         ? state + 1
+ *         : state
+ *
+ * const schema = { type: 'number', minimum: 0 }
+ *
+ * export const age = withSchemaValidation( schema, age )
+ *
+ * // ageReducer( -5, { type: DESERIALIZE } ) === -5
+ * // age( -5, { type: DESERIALIZE } ) === 0
+ * // age( 23, { type: DESERIALIZE } ) === 23
+ *
+ * @param {object} schema JSON-schema description of state
+ * @param {function} reducer normal reducer from ( state, action ) to new state
+ * @returns {function} wrapped reducer handling validation on DESERIALIZE
+ */
+export const withSchemaValidation = ( schema, reducer ) => ( state, action ) => {
+	if ( DESERIALIZE === action.type ) {
+		return state && isValidStateWithSchema( state, schema )
+			? state
+			: reducer( undefined, { type: '@@dummy/INIT' } );
```

Comment:
Yeah I figured, since we just want default state here. Perhaps a comment here then at least? I'm not sure how many folks know about the `@@` prefix or other redux internal events.